package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -930,7 +930,8 @@ def _get_timestep():
             timestep = _get_timestep()
             self.timesteps.append(timestep)
 
-            pm.echo(ts.frame)
+            progress = (ts.frame - (self.traj_slice.start or 0)) // (self.traj_slice.step or 1)
```

Comment:
Isn't this calculation similar to the following for loop change?

```python
for i, ts in enumerate(self.u.trajectory[self.traj_slice]):
    pm(i)
```   

Then I would prefer the later. It's just a bit more standard way to have the current loop count in python. This also works with your examples.